frontend/src/metabase/lib/data_grid.js

```diff
@@ -375,6 +383,23 @@ function updateValueObject(row, indexes, seenValues, collapsedSubtotals = []) {
     if (seenValue === undefined) {
       seenValue = { value, children: [], isCollapsed };
       currentLevelSeenValues.push(seenValue);
+      const sortOrder = getIn(columnSettings, [index, COLUMN_SORT_ORDER]);
+      if (sortOrder) {
+        currentLevelSeenValues.sort((a, b) => {
```

Comment:
This block is a bit tricky to follow along. A potential follow-up will be a more self-contained version of the block (a smaller pure function?), along with the corresponding unit tests.

Reply:
Yeah good call. I'll extract that now. The branches (strings with accents, numbers) are covered through the unit tests of `multiLevelPivot`, but pulling out a distinct function would make the code more clear.